src/Core/Module/ModuleCollection.php

```diff
@@ -138,113 +137,113 @@ public function offsetUnset($offset)
     {
-        return isset($this->addons[$key]) || array_key_exists($key, $this->addons);
+        return isset($this->modules[$key]) || array_key_exists($key, $this->modules);
     }
 
     /**
-     * Returns true if the addon is found in the collection.
+     * Returns true if the module is found in the collection.
      *
-     * @param Addon $addon the addon
+     * @param ModuleInterface $module the module
      *
      * @return bool
      */
-    public function contains(Addon $addon)
+    public function contains(ModuleInterface $module)
     {
-        return in_array($addon, $this->addons, true);
+        return in_array($module, $this->modules, true);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function indexOf(Addon $addon)
+    public function indexOf(ModuleInterface $module)
     {
-        return array_search($addon, $this->addons, true);
+        return array_search($module, $this->modules, true);
     }
 
     /**
      * {@inheritdoc}
      */
     public function get($key)
     {
-        return $this->addons[$key] ? $this->addons[$key] : null;
+        return $this->modules[$key] ?: null;
     }
 
     /**
      * {@inheritdoc}
      */
     public function getKeys()
     {
-        return array_keys($this->addons);
+        return array_keys($this->modules);
     }
 
     /**
      * {@inheritdoc}
      */
     public function getValues()
     {
-        return array_values($this->addons);
+        return array_values($this->modules);
     }
 
     /**
-     * Add an Addon with a specified key in the collection.
+     * Add a module with a specified key in the collection.
      *
      * @param mixed $key the key
-     * @param Addon $addon the specified addon
+     * @param ModuleInterface $module the specified module
      */
-    public function set($key, Addon $addon)
+    public function set($key, ModuleInterface $module)
     {
-        $this->addons[$key] = $addon;
+        $this->modules[$key] = $module;
     }
 
     /**
-     * Add an Addon in the collection.
+     * Add a Module in the collection.
      *
-     * @param Addon $addon the specified addon
+     * @param ModuleInterface $module the specified module
      *
      * @return bool
      */
-    public function add(Addon $addon)
+    public function add(ModuleInterface $module)
     {
-        $this->addons[] = $addon;
+        $this->modules[] = $module;
 
         return true;
     }
 
     /**
-     * Remove an addon from the collection by key.
+     * Remove a module from the collection by key.
      *
      * @param int|string $key
      *
-     * @return bool|null true if the addon has been found and removed
+     * @return bool|null true if the module has been found and removed
      */
     public function removeByKey($key)
     {
-        if (!isset($this->addons[$key]) && !array_key_exists($key, $this->addons)) {
+        if (!isset($this->modules[$key]) && !array_key_exists($key, $this->modules)) {
```

Comment:
Type is safe here, no need to double check.
```suggestion
        if (!isset($this->modules[$key])) {
```